comp/core/telemetry/telemetry.go

```diff
@@ -0,0 +1,200 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package telemetry
+
+import (
+	"net/http"
+
+	promOtel "go.opentelemetry.io/otel/exporters/prometheus"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/otel/metric"
+	sdk "go.opentelemetry.io/otel/sdk/metric"
+)
+
+// TODO (components): Remove the global and move this into `newTelemetry` after all telemetry is migrated to the component
+var (
+	registry = newRegistry()
+	provider = newProvider(registry)
```

Comment:
Can you explain why these variables have to be global and not injected in the constructor? It is not so obvious when looking at the code.

Reply:
This is so `GetCompatComponent` can return an instance of the component without FX. This is to support legacy code. 

Once the legacy code is no-longer needed, we can remove this and the registry and provider state will be managed by the components lifecycle. The `TODO` above should capture this. 

Unlike the log component that is a shim for the global logger, I took the opposite approach here where the legacy telemetry code talks to the component under the hood. This should make removing the old code much simpler when the time comes.